File: Exercise/dice-game/java/DiceGame.java
public class DiceGame {
        private Dice dice1 = new Dice();
        private Dice dice2 = new Dice();

        private final static int WIN_SUM = 7;

        public void startGame() {
                dice1.roll();
                showFaceValue(dice1.getFaceValue());

                dice2.roll();
                showFaceValue(dice2.getFaceValue());

                showGameResult();
        }

        public boolean isPlayerWin() {
                return dice1.getFaceValue().toInteger() + dice2.getFaceValue().toInteger() == WIN_SUM;
        }

        private void showFaceValue(FaceValue faceValue) {
                System.out.println("Face value: " + faceValue.toInteger());
        }

        private void showGameResult() {
                if (isPlayerWin()) {
                        System.out.println("win");
                } else {
                        System.out.println("loss");
                }
        }
}

File: Exercise/dice-game/python/dicegame.py
from dice import Dice

class DiceGame(object):
    win_number = 7
    
    def __init__(self):
        self._dice1 = Dice()
        self._dice2 = Dice()
        self._is_player_win = False

    def check_game_result(self):
        if self._dice1.face_value() + self._dice2.face_value() == self.win_number:
            self._is_player_win = True
        else:
            self._is_player_win = False

    def start_game(self):
        self._dice1.roll()
        self.show_face_value(self._dice1.face_value())
        self._dice2.roll()
        self.show_face_value(self._dice2.face_value())
        self.check_game_result()
        self.show_game_result()

    def show_face_value(self, face_value):
        print(face_value)

    def show_game_result(self):
        if self._is_player_win:
            print("win")
        else:
            print("loss")
